### PR TITLE
upgrade path feature for TimescaleDB 0.5.0 to 0.8.2

### PR DESCRIPTION
**Pull request by VineethReddy02**

### CODE REVIEW

File: cli/cmd/helm/upgrade.go [hunk 1]
```diff
@@ -179,3 +188,187 @@ func upgradeTobs(cmd *cobra.Command, args []string) error {
...
+	}
+
+	dVersion, err := utils.ParseVersion(deployedChartVersion, 3)
+	if err != nil {
+		return fmt.Errorf("failed to parse deployed helm chart version %w", err)
+	}
+
+	fmt.Println(deployedChartVersion, nVersion, dVersion)
+	switch {
+	case deployedChartVersion <= "0.2.2" && nVersion > dVersion:
```
Comment (cevian) [inline cli/cmd/helm/upgrade.go:205]: these comparisons aren't right, you can't use string comparison. also I think you want nVersion > cutVersion not nVersion > dVsersion
```suggestion
	cutVersion, err := utils.ParseVersion("0.2.2", 3)
	if err != nil {
		return fmt.Errorf("failed to parse deployed helm chart version %w", err)
	}
	fmt.Println(deployedChartVersion, nVersion, dVersion)
	switch {
	case dVersion <= cutVersion && nVersion > cutVersion:
```

File: cli/cmd/helm/upgrade.go [hunk 2]
```diff
@@ -179,3 +188,187 @@ func upgradeTobs(cmd *cobra.Command, args []string) error {
...
+	if err != nil {
+		return fmt.Errorf("failed to parse latest helm chart version %w", err)
+	}
+
+	dVersion, err := utils.ParseVersion(deployedChartVersion, 3)
+	if err != nil {
+		return fmt.Errorf("failed to parse deployed helm chart version %w", err)
+	}
+
+	fmt.Println(deployedChartVersion, nVersion, dVersion)
```
Comment (cevian) [inline cli/cmd/helm/upgrade.go:203]: get rid of this

File: cli/cmd/helm/upgrade.go [hunk 3]
```diff
@@ -179,3 +188,187 @@ func upgradeTobs(cmd *cobra.Command, args []string) error {
...
+	fmt.Println("Migrating backup configuration")
+
+	newData := make(map[string][]byte)
+	for key, value := range s.Data {
+		ok := strings.Contains(key, "-s3-")
+		if ok {
+			newKey := strings.Replace(key, "-", "_", -1)
+			newKey = strings.ToUpper(newKey)
+			newKey = "PGBACKREST_" + newKey
+			newData[newKey] = value
```
Comment (cevian) [inline cli/cmd/helm/upgrade.go:352]: should the value be base64 encoded?

Comment (VineethReddy02) [reply cli/cmd/helm/upgrade.go]: encoding is done internally. We just pass the data in the format of bytes.

File: cli/cmd/helm/upgrade.go (hunk 1 above)
Comment (VineethReddy02) [reply cli/cmd/helm/upgrade.go]: So here the condition is
`dVersion <= version0_2_2 && nVersion > dVersion` so I expect the deployed version to be <= 0.2.2 and new Version to be greater than deployed Version.`

The deployed version in users env can be 0`.1.1` too.

**Review by cevian — APPROVED** [review]

**Pull request merged**